internal/locald/sandboxmanager/operator_info_updater.go

```diff
@@ -0,0 +1,80 @@
+package sandboxmanager
+
+import (
+	"context"
+	"log/slog"
+	"sync"
+	"time"
+
+	commonapi "github.com/signadot/cli/internal/locald/api"
+	tunapiv1 "github.com/signadot/libconnect/apiv1"
+	tunapiclient "github.com/signadot/libconnect/common/apiclient"
+)
+
+type operatorInfoUpdater struct {
+	sync.Mutex
+
+	log          *slog.Logger
+	operatorInfo *commonapi.OperatorInfo
+	isRunning    bool
+}
+
+func (oiu *operatorInfoUpdater) Get() *commonapi.OperatorInfo {
+	oiu.Lock()
+	defer oiu.Unlock()
+	return oiu.operatorInfo
+}
+
+func (oiu *operatorInfoUpdater) Reset() {
+	oiu.Lock()
+	defer oiu.Unlock()
+	oiu.operatorInfo = nil
+}
+
+func (oiu *operatorInfoUpdater) Reload(ctx context.Context, tunAPIClient tunapiclient.Client, force bool) {
+	oiu.Lock()
+	defer oiu.Unlock()
+
+	if oiu.isRunning {
+		// nothing to do here
+		return
+	}
+	if !force && oiu.operatorInfo != nil {
+		// we already have the operator info
+		return
+	}
+	// reload the data
+	oiu.isRunning = true
```

Comment:
Can we put this inside of `reload` (maybe via new method setIsRunning w/ lock) so that the invariants for the func `reload` are contained within its definition?

Reply:
It's done in https://github.com/signadot/cli/pull/95/commits/d3f865c7fdde20beb14e4fad747b45621f41bdb4.